changelogs/fragments/10011-github_deploy_key-check-key-present.yml

```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - "github_deploy_key - fix: check that key really exists on 422 to avoid masking other errors (https://github.com/ansible-collections/community.general/issues/6718, https://github.com/ansible-collections/community.general/pull/10011)"
```

Comment:
```suggestion
  - "github_deploy_key - check that key really exists on 422 to avoid masking other errors (https://github.com/ansible-collections/community.general/issues/6718, https://github.com/ansible-collections/community.general/pull/10011)."
```